src/rateEngine/billingDeterminants/SurchargeAsPercent.ts

```diff
@@ -0,0 +1,36 @@
+import BillingDeterminants, { BillingDeterminantsUnits } from './_BillingDeterminants';
+import RateElement, { RateElementClassification } from '../RateElement';
+import LoadProfile from '../LoadProfile';
+
+export interface SurchargeAsPercentArgs {
+  rateElement: RateElement;
+  percent: number;
+}
+
+class SurchargeAsPercent extends BillingDeterminants {
+  private _rateElement: RateElement;
+  private _percent: number;
+
+  rateElementType = 'Surcharge';
+  rateElementClassification = RateElementClassification.ENERGY;
+  units = BillingDeterminantsUnits.DOLLARS;
+
+  constructor({rateElement}: SurchargeAsPercentArgs) {
+    super();
+
+    this._rateElement = rateElement;
+  }
+
+  calculate(): Array<number> {
```

Comment:
The billing determinant calculation for the surcharges (as percents) ends up being simple. It just sums the monthly costs for all of the rate components and returns dollars / month. The charge (which is defined by the `RateComponentsFactory`) ends up being the percentage * 1 dollar.

I think this will be nice for auditing because billing determinant for each of these rate components should equal the cost of its corresponding rate element.

"billing determinant" is a surprisingly easy concept to implement, given that it can represent any arbitrary unit of data!